src/modules/dashboard/components/Members/Members.tsx

```diff
@@ -166,91 +211,84 @@ const Members = ({ colony: { colonyAddress, colonyName }, colony }: Props) => {
    * of the results, with the next waiting in line for a callback, but the current
    * timeframe doesn't allow for this, so I guess this is the "best" we can do for now
    */
-  const paginatedContributors = contributors.slice(
-    0,
-    ITEMS_PER_PAGE * dataPage,
-  );
+  const paginatedContributors =
+    contributors?.slice(0, ITEMS_PER_PAGE * dataPage) || [];
 
-  const paginatedWatchers = watchers.slice(0, ITEMS_PER_PAGE * dataPage);
+  const paginatedWatchers = watchers?.slice(0, ITEMS_PER_PAGE * dataPage) || [];
 
   return (
     <div className={styles.main}>
-      <div className={styles.titleContainer}>
-        <Heading
-          text={MSG.title}
-          textValues={{
-            domainLabel: selectedDomain
-              ? selectedDomain.name
-              : ALLDOMAINS_DOMAIN_SELECTION.name,
-          }}
-          appearance={{ size: 'medium', theme: 'dark' }}
-        />
-        <Form
-          initialValues={{ filter: currentDomainId.toString() }}
-          onSubmit={() => {}}
-        >
-          <Select
-            appearance={{
-              alignOptions: 'right',
-              theme: 'alt',
-            }}
-            elementOnly
-            label={MSG.labelFilter}
-            name="filter"
-            onChange={setFieldValue}
-            options={domainSelectOptions}
+      <MembersTitle
+        currentDomainId={currentDomainId}
+        handleDomainChange={handleDomainChange}
+        domainSelectOptions={domainSelectOptions}
+        searchValue={searchValue}
+        setSearchValue={setSearchValue}
+        handleSearch={handleSearch}
+      />
+      <div>
+        {/* This needs attention!!!! - ready to be refactored in another PR
```

Comment:
Since Sayan merged his PR already, please, rebase it and refactor as needed. It's already possible to refactor it in this PR.